apps/acrs/behaviours/resume-form-session.js

```diff
@@ -0,0 +1,203 @@
+    let cleanList = Object.keys(req.sessionModel.attributes);
+    const leaveFields = SESSION_DEFAULTS.fields.concat(['csrf-secret', 'user-cases']);
+    cleanList = cleanList.filter(item => leaveFields.indexOf(item) === -1);
+
+    req.sessionModel.unset(cleanList);
+    req.sessionModel.set('steps', SESSION_DEFAULTS.steps);
+  }
+  // GET lifecycle
+  configure(req, res, next) {
+    // reset session but keeping login fields, csrf and preexisting cases garnered from API
+    this.cleanSession(req);
+    // skip requesting data service api when running in local and test mode
+    if (config.env === 'local' || config.env === 'test') {
+      return super.configure(req, res, next);
+    }
+    return this.getCases(req, res, next);
+  }
+
+  getCases(req, res, next) {
+    const email = req.sessionModel.get('user-email');
+    const idType = req.sessionModel.get('id-type');
+
+    return axios.get(baseUrl + encodeEmail(email))
+      .then(response => {
+        const sessionCases = req.sessionModel.get('user-cases') || [];
+        const unsubmittedCases = _.filter(response.data, record => !record.submitted_at &&
+                                                                    record.session['id-type'] === idType);
+        const parsedBody = this.parseCasesSessions(unsubmittedCases);
+        const filteredSessionCases = sessionCases.filter(record => record['id-type'] === idType);
+        const cases = _.unionBy(parsedBody, filteredSessionCases, 'id');
+
+        const uan = req.sessionModel.get('uan');
+        const brp = req.sessionModel.get('brp');
+        let isSameCase = '';
+        let idNumber = '';
+
+        if (idType === 'brp') {
+          isSameCase = _.get(cases, "[0].session['brp']") === brp;
+          idNumber = brp;
+        }else{
+          isSameCase = _.get(cases, "[0].session['uan']") === uan;
+          idNumber = uan;
+        }
+
+
+        const singleCase = cases.length < 2;
+
+        const noCaseOrSameCase = !cases[0] || isSameCase;
+        const multipleCasesInSession = _.get(req.sessionModel.get('user-cases'), 'length') > 1;
+
+        if (singleCase && noCaseOrSameCase && !multipleCasesInSession) {
+          if (cases[0]) {
+            this.setupSession(req, cases[0].session);
+          }
+          req.sessionModel.set('multipleCases', false);
+          return res.redirect(super.getNextStep(req, res, next));
+        }
+
+        const filteredCases = this.addCasesToSession(req, cases, idNumber);
+        this.setupRadioButtons(req, filteredCases.filter(form => form.id));
+
+        return super.configure(req, res, next);
+      })
+      .catch(next);
+  }
+
+  addCasesToSession(req, cases, idNumber) {
+    const idType = req.sessionModel.get('id-type');
+    let caseAlreadyInSession = '';
+
+    if (idType === 'brp') {
+      caseAlreadyInSession = cases.find(obj => obj.session.brp === idNumber);
+    }else{
+      caseAlreadyInSession = cases.find(obj => obj.session.uan === idNumber);
+    }
```

Comment:
you could probably do `const caseAlreadyInSession = cases.find(obj => obj.session[idType] === idNumber;`
But this would work fine already I think

Reply:
declaring a const variable will make it immutable, so in this case we have the variable assignment in the If block

Reply:
I'm saying you don't need the if block if you don't want it - you can simplify the code, but it will work equally as you have done it